### PR TITLE
Issue7 logout

### PR DESCRIPTION
**Pull request by mohes7489**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -133,7 +158,7 @@
 				<plugin>
 					<groupId>org.jooq</groupId>
 					<artifactId>jooq-codegen-maven</artifactId>
-					<version>3.19.13</version>
+					<version>3.19.11</version>
```
Comment (williamirva) [inline pom.xml:161]: What was the reason to change the JOOQ-version?

File: src/main/java/com/apuope/apuope_re/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+public class SecurityConfig {
+
+    @Autowired
+    private MyUserDetailService userDetailsService;
+
+    @Autowired
+    private JWTFilter jwtFilter;
+
+
+    @Bean
```
Comment (williamirva) [inline src/main/java/com/apuope/apuope_re/config/SecurityConfig.java:36]: I am not sure if the bean annotation is required in methods. Have you tried to delete them?

File: src/main/java/com/apuope/apuope_re/services/JWTService.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public String generateToken(String email) {
+        Map<String, Object> claims = new HashMap<>();
+        return Jwts.builder()
+                .claims()
+                .add(claims)
+                .subject(email)
+                .issuedAt(new Date(System.currentTimeMillis()))
+                .expiration(new Date(System.currentTimeMillis()+ 1000 * 60 * 60 * 24))
```
Comment (williamirva) [inline src/main/java/com/apuope/apuope_re/services/JWTService.java:31]: You could create global variable for the expiration time

File: src/main/java/com/apuope/apuope_re/services/LogInService.java [hunk 4]
```diff
@@ -5,30 +5,40 @@
...
     public LogInService(DSLContext dslContext, UserRepository userRepository) {
         this.dslContext = dslContext;
         this.userRepository = userRepository;
+        this.passwordEncoder = new BCryptPasswordEncoder(10);
     }
 
     public ResponseData<String> validateUser(UserCredentials userCredentials) {
         Optional<UsersRecord> userOpt =
-                userRepository.findVerifiedUserByEmail(userCredentials.getEmail(), this.dslContext);
+                userRepository.findByEmail(userCredentials.getEmail(), this.dslContext);
```
Comment (williamirva) [inline src/main/java/com/apuope/apuope_re/services/LogInService.java:30]: This should be discussed, do we want to allow user to login without verifying the email?

File: src/main/java/com/apuope/apuope_re/controller/testController.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    @GetMapping("/test")
+    public String test() {
+        return "Test...";
+    }
+
+    @GetMapping("/test2")
+    public String test2() {
+        return "Test2...";
+    }
+}
```
Comment (williamirva) [inline src/main/java/com/apuope/apuope_re/controller/testController.java:19]: I guess you have used this for testing the JWT, but it should not be included to merge.

File: pom.xml (hunk 1 above)
Comment (mohes7489) [reply pom.xml:161]: I needed to change it, my IDE was giving me a mismatch warning with the version number.

File: src/main/java/com/apuope/apuope_re/controller/testController.java (hunk 5 above)
Comment (mohes7489) [reply src/main/java/com/apuope/apuope_re/controller/testController.java:19]: yes I agreed, I just left them for testing purposes

File: src/main/java/com/apuope/apuope_re/services/JWTService.java (hunk 3 above)
Comment (mohes7489) [reply src/main/java/com/apuope/apuope_re/services/JWTService.java:31]: yes I could, if that make the code readability easier for everyone, I just thought is neat to do it like that.

**Pull request closed without merging**